spec/unit/hanami/settings_spec.rb

```diff
@@ -41,6 +44,32 @@
         /(database_url: nope to database).*(redis_url: nope to redis)/m,
       )
     end
+
+    it "collects errors for missing settings failing their constructors" do
+      settings_class = Class.new(described_class) do
+        setting :database_url, constructor: ->(_v) { raise "nope to database" }
+      end
+
+      store = {}
+
+      expect { settings_class.new(store) }.to raise_error(
+        described_class::InvalidSettingsError,
+        /database_url: nope to database/m,
+      )
+    end
+
+    it "finalizes the config" do
+      settings_class = Class.new(described_class) do
+        setting :database_url
+      end
+
+      store = {database_url: "postgres://localhost/test_app_development"}
+
+      settings = settings_class.new(store)
+
+      expect(settings.config).to be_frozen
+      expect { settings.database_url = "new" }.to raise_error(Dry::Configurable::FrozenConfig)
```

Comment:
Jus a heads up: I'll need to change this to FrozenConfigError once we make the next dry-configurable release.